src/MonthSelect.php

```diff
@@ -4,24 +4,84 @@
 
 namespace Laminas\Filter;
 
+use function filter_var;
+use function is_array;
+use function is_numeric;
+use function sprintf;
+
+use const FILTER_VALIDATE_INT;
+
 /**
  * @psalm-type Options = array{
  *     null_on_empty?: bool,
  *     null_on_all_empty?: bool,
- *     ...
  * }
  * @psalm-type InputArray = array{
- *       year: numeric,
- *       month: numeric,
+ *       year: int,
+ *       month: int,
  *  }
- * @template TOptions of Options
- * @template-extends AbstractDateDropdown<TOptions, InputArray>
+ * @template TInput of array<array-key, numeric>
+ * @implements FilterInterface<string>
  */
-final class MonthSelect extends AbstractDateDropdown
+final class MonthSelect implements FilterInterface
 {
-    /**
-     * Year-Month
-     */
-    protected string $format      = '%2$s-%1$s';
-    protected int $expectedInputs = 2;
+    private readonly bool $returnNullIfAnyFieldEmpty;
+    private readonly bool $returnNullIfAllFieldsEmpty;
+
+    /** @param Options $options */
+    public function __construct(array $options = [])
+    {
+        $this->returnNullIfAnyFieldEmpty  = $options['null_on_empty'] ?? false;
+        $this->returnNullIfAllFieldsEmpty = $options['null_on_all_empty'] ?? false;
+    }
+
+    public function __invoke(mixed $value): mixed
+    {
+        return $this->filter($value);
+    }
+
+    /** @psalm-suppress InvalidReturnType */
```

Comment:
Not sure why this suppression is necessary, possibly, we should make `@implements FilterInterface<string|null>` instead of just `string`